src/components/icon/icon.tsx

```diff
@@ -134,11 +136,27 @@ export class Icon {
   @Watch('icon')
   loadIcon() {
     if (Build.isBrowser && this.isVisible) {
+      if (!parser) {
+        /**
+         * Create an instance of the DOM parser. This creates a single
+         * parser instance for the entire app, which is more efficient.
+         */
+        parser = new DOMParser();
+      }
       const url = getUrl(this);
+
       if (url) {
         if (ioniconContent.has(url)) {
           // sync if it's already loaded
           this.svgContent = ioniconContent.get(url);
+        } else if (url.startsWith('data:')) {
+          const doc = parser.parseFromString(url, 'text/html');
+          const svgEl = doc.body.querySelector('svg');
```

Comment:
Why do we query for an `svg` element in the body?

Reply:
This was inspired from Shoelace. It looks like we could query directly from the constructed document (`doc`), instead of the body element. Would you prefer that change?

Reply:
Oh 🤦 I misunderstood that as querying from the main document not the parsed document. The current approach looks good to me.